server/src/main/java/io/druid/server/coordinator/DruidCluster.java

```diff
@@ -113,9 +115,11 @@ private void addHistorical(ServerHolder serverHolder)
 
   public Collection<ServerHolder> getAllServers()
   {
-    return historicals.values().stream()
-                      .flatMap(Collection::stream)
-                      .collect(() -> realtimes, Set::add, Set::addAll);
+    final List<ServerHolder> allServers = historicals.values().stream()
```

Comment:
Suggested more old-style to avoid list reallocation:
```java
int historicalServers = historicals.values().stream().mapToInt(Collection::size).sum();
int allServers = historicalServers + realtimes.size();
List<ServerHolder> result = new ArrayList<>(allServers);
historicals.values().forEach(tierServers -> result.addAll(tierServers));
result.addAll(realtimes);
return result;
```